### DISCUSSION TITLE
Only first table row expanded

### DISCUSSION THREAD

**Discussion by JustEp1c** [ant-design-blazor/ant-design-blazor#3690] [Q&A]
Hello! Is there a way to initially expand only first row in a table? I know that there are two attributes ( `DefaultExpandAllRows` and `DefaultExpandMaxLevel` ) which collectively can help expand all rows on whole level on specified depth, but that's not what i'm looking for.
The best I could achieve with these attributes is that:

![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/85038502/949121b0-75a4-4536-ae48-5d219bd12474)

But i need that:

![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/85038502/82d0bc14-27f8-4a91-939e-a2322b38550e)

**Comment by ElderJames** [comment]
Thanks for contacting us @JustEp1c , you can use `OnRow` parameter to custom the RowData and row parameters sot that allow to set up `Expanded`:

```
<Table TItem="Data" DataSource="@data" OnRow="OnRow">
...

@code {
    Dictionary<string, object> OnRow(RowData<Data> row)
    {
        if (row.Data.Key == "1")
        {
            row.Expanded = true;
        }

        return new();
     }

}
```

**Reply by JustEp1c** [reply]
Thank you, that is the answer to my question, but with this solution i can't collapse the row anymore. What should I do?
